service/builder.go

```diff
@@ -53,12 +54,14 @@ func (b *AgentServiceBuilder) Build(ctx context.Context) (service.Service, conte
 func (b *AgentServiceBuilder) buildAgent() (*agent.LogAgent, error) {
 	agentBuilder := agent.NewBuilder(b.logger)
 
-	if b.config != nil {
-		agentBuilder = agentBuilder.WithConfig(&agent.Config{
-			Pipeline: b.config.Pipeline,
-		})
+	if b.config == nil {
+		return nil, errors.New("config cannot be nil")
```

Comment:
Nit: if this is a customer visible error we probably should say `cannot be nil`. That may not mean anything to the person setting it up.

Thinking about this though would this basically means you didn't set a config file which means you probably failed way before this? If that's true leave as is. Only someone using this as a library would hit this error in that case.

Reply:
Yeah, this is a scenario that a customer shouldn't see. If you don't provide a config, you get this error when it can't find the default: "Failed to load config: could not read config file: open /Users/brandon/stanza/config.yaml: no such file or directory"

Basically, if this happens, it's programmer error.